### PR TITLE
initial commit for skeleton ansible for jenkins worker config

### PR DESCRIPTION
**Pull request by shaneknapp**
### What changes were proposed in this pull request?
this is the skeleton of the ansible used to configure jenkins workers in the riselab/apache spark build system

### Why are the changes needed?
they are not needed, but will help the community understand how to build systems to test multiple versions of spark, as well as propose changes that i can integrate in to the "production" riselab repo.  since we're sunsetting jenkins by EOY 2021, this will potentially be useful for migrating the build system.

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
ansible-lint and much wailing and gnashing of teeth.

### CODE REVIEW

**Comment by shaneknapp** [comment]
i'm going to need to do some more refactoring, but not much...

File: dev/ansible-for-test-node/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+* add this to your `~/.ansible.cfg`:
+```[defaults] host_key_checking = False```
+
+#### fire ansible cannon!
+`ansible-playbook -u <service account> deploy-jenkins-worker.yml -i <ansible-style-hosts-file> -k -b -K` 
+
+tips:
+* if you are installing more than a few workers, it's best to run the playbook on smaller (2-3) batches at a time.  this way it's easier to track down errors, as ansible is very noisy.
+* when you encounter an error, you should comment out any previously-run plays and tasks.  this saves time when debugging, and let's you easily track where you are in the process.
+* `apt-get remove <application_name>` and `apt-get purge <package-name>` are your friends
```
Comment (shaneknapp) [inline dev/ansible-for-test-node/README.md:25]: TODO:  explain more the scope of the ansible in the bigger picture of the build system

**Comment by Yikun** [comment]
@shaneknapp Hi, shane, do you have any plan to append Arm node [1] setup related configuration on your patch? Thanks

[1] https://amplab.cs.berkeley.edu/jenkins/label/spark-arm/

**Comment by shaneknapp** [comment]
going to merge this in to master later on today

**Comment by shaneknapp** [comment]
test this please

**Comment by shaneknapp** [comment]
test this please

**Pull request closed without merging**